packages/agents/src/tests/index.test.ts

```diff
@@ -0,0 +1,672 @@
+      } else {
+        expect(context).toBeUndefined();
+      }
+    });
+  });
+
+  describe("Route Pattern Matching", () => {
+    it("should return null for non-agent paths", async () => {
+      const nonAgentPaths = [
+        "/other/path",
+        "/api/endpoint",
+        "/",
+        "/agents",
+        "/agents/"
+      ];
+
+      for (const path of nonAgentPaths) {
+        const request = new Request(`http://localhost${path}`, {
+          method: "GET"
+        });
+
+        const response = await routeAgentRequest(request, env);
+        expect(response).toBeNull();
+      }
+    });
+
+    it("should detect valid agent paths", async () => {
+      const validPaths = [
+        "/agents/TestAgent/instance",
+        "/agents/test-agent/room-123",
+        "/agents/EmailAgent/user@example.com"
+      ];
+
+      for (const path of validPaths) {
+        const request = new Request(`http://localhost${path}`, {
+          method: "OPTIONS",
+          headers: {
+            Origin: "http://localhost:3000"
+          }
+        });
+
+        const response = await routeAgentRequest(request, env, { cors: true });
+
+        expect(response).not.toBeNull();
+        expect(response!.status).toBe(200);
+        expect(response!.headers.get("Access-Control-Allow-Origin")).toBe("*");
+      }
+    });
+  });
+
+  describe("Advanced Integration", () => {
+    it("should handle kebab-case to CamelCase agent name resolution", async () => {
+      const request = new Request(
+        "http://localhost/agents/test-agent/my-instance",
+        {
+          method: "GET"
+        }
+      );
+
+      const response = await routeAgentRequest(request, env);
+
+      expect(response).toBeDefined();
+      expect(response).not.toBeNull();
+    });
+
+    it("should support location hints", async () => {
+      const locationHints: DurableObjectLocationHint[] = [
+        "wnam",
+        "enam",
+        "weur",
+        "eeur",
+        "apac"
+      ];
+
+      for (const hint of locationHints) {
+        const agent = await getAgentByName(env.TEST_AGENT, `location-${hint}`, {
+          locationHint: hint
+        });
+        expect(agent).toBeDefined();
+      }
+    });
+  });
+
+  describe("Agent State Management", () => {
+    it("should persist state between requests", async () => {
+      const agent = await getAgentByName(env.TEST_AGENT, "state-persist-test");
+
+      const setResponse = await agent.fetch(
+        new Request("http://localhost/setState", {
+          method: "POST",
+          headers: { "Content-Type": "application/json" },
+          body: JSON.stringify({ count: 5, name: "test" })
+        })
+      );
+
+      expect(setResponse.status).toBe(200);
+      const setResult = (await setResponse.json()) as { success: boolean };
+      expect(setResult.success).toBe(true);
+
+      const getResponse = await agent.fetch(
+        new Request("http://localhost/getState")
+      );
```

Comment:
we actually want to evict the DO before making the second request here. I think there's a way to do that, let's chat about it offline, can't recollect atm